build/configs/rtl8721csm/rtl8721csm_download.sh

```diff
@@ -44,17 +44,33 @@ CUR_PATH=$(dirname "$CURDIR")
 TOOL_PATH=${CUR_PATH}/../../tools/amebad
 IMG_TOOL_PATH=${TOOL_PATH}/image_tool
 BIN_PATH=${CUR_PATH}/../../output/bin
+TOP_PATH=${CUR_PATH}/../../..
+OS_PATH=${TOP_PATH}/os
 SUDO=sudo
 
+CONFIG=${OS_PATH}/.config
+source ${CONFIG}
+
 cp -p ${BIN_PATH}/km0_boot_all.bin ${IMG_TOOL_PATH}/km0_boot_all.bin
 cp -p ${BIN_PATH}/km4_boot_all.bin ${IMG_TOOL_PATH}/km4_boot_all.bin
 cp -p ${BIN_PATH}/km0_km4_image2.bin ${IMG_TOOL_PATH}/km0_km4_image2.bin
+if [[ "${CONFIG_APP_BINARY_SEPARATION}" == "y" ]];then
+	cp -p ${BIN_PATH}/rtl8721csm_smartfs.bin ${IMG_TOOL_PATH}/rtl8721csm_smartfs.bin
+fi
 
 cd ${IMG_TOOL_PATH}
 
 echo "Start Download"
-$SUDO ./amebad_image_tool /dev/$PORT
+source ${CONFIG}
+if [[ "${CONFIG_APP_BINARY_SEPARATION}" != "y" ]];then
+	$SUDO ./amebad_image_tool /dev/$PORT
+else
+	$SUDO ./amebad_image_tool /dev/$PORT 0x08000000 0x08004000 0x08006000 0x08180000
```

Comment:
The offset of flash should not be hard-coded. Let's add bootloaders and kernel  partitions in the config and then calculate the offset.

Reply:
Modified. Kindly check